### ISSUE TITLE
'Object not found' error after shard switched to disabled mode

### ISSUE THREAD

**Issue by anikeev-yadro** [nspcc-dev/neofs-node#1883]
## Expected Behavior
For object read operations. After we switched shard to disabled mode on the fly we expect to nothing to be happined. Because we have second copy on the other node.

## Current Behavior
After we switched shard to disabled mode on the fly we had erros 'Object not found' on the object read operations.

## Steps to Reproduce (for bugs)
1. Create containers and preload objects
```
admin@perf-load-01:~/xk6$ ./scenarios/preset/preset_grpcREP2-Russia.py --size 10000 --containers 5 --out grpc_50kb.json --endpoint 172.26.161.204:8080 --preload_obj 5
Namespace(size='10000', containers='5', out='grpc_50kb.json', preload_obj='5', endpoint='172.26.161.204:8080', update=None)
Create containers: 5
Create containers: Completed
 > Containers: ['8YuaHZJgfKZbwBJLVDR331gKfQyZg1RG99gxsXUtVoxx', '9mt3SyAzag1jFczfAUyY6fS3B91kTqRy1vwH8WtNb7Yr', '9LrqnwWBpfkYQ8SuQexaUEXyqLE5yuzp7i3vWipxzMRr', '331CCw5pEseWszpCuDLiFoeCaqiATzEpMzHqC5hVa4vA', '3MkVppadVwmhncqYsHmbiuxgdEUBiFGjZRG2fEixCeFD']
Upload objects to each container: 5
 > Create random payload: Completed
 > Upload objects for container 8YuaHZJgfKZbwBJLVDR331gKfQyZg1RG99gxsXUtVoxx
 > Upload objects for container 8YuaHZJgfKZbwBJLVDR331gKfQyZg1RG99gxsXUtVoxx: Completed
 > Upload objects for container 9mt3SyAzag1jFczfAUyY6fS3B91kTqRy1vwH8WtNb7Yr
 > Upload objects for container 9mt3SyAzag1jFczfAUyY6fS3B91kTqRy1vwH8WtNb7Yr: Completed
 > Upload objects for container 9LrqnwWBpfkYQ8SuQexaUEXyqLE5yuzp7i3vWipxzMRr
 > Upload objects for container 9LrqnwWBpfkYQ8SuQexaUEXyqLE5yuzp7i3vWipxzMRr: Completed
 > Upload objects for container 331CCw5pEseWszpCuDLiFoeCaqiATzEpMzHqC5hVa4vA
 > Upload objects for container 331CCw5pEseWszpCuDLiFoeCaqiATzEpMzHqC5hVa4vA: Completed
 > Upload objects for container 3MkVppadVwmhncqYsHmbiuxgdEUBiFGjZRG2fEixCeFD
 > Upload objects for container 3MkVppadVwmhncqYsHmbiuxgdEUBiFGjZRG2fEixCeFD: Completed
Upload objects to each container: Completed
Result:
 > Total Containers has been created: 5.
 > Total Objects has been created: 25.
```
2. Start k6 load with read objects
```
admin@perf-load-01:~/xk6$ date;./k6 run -e PROFILE=50:240 -e WRITE_OBJ_SIZE=10000 -e CLIENTS=50 -e NODES=172.26.161.204:8080 -e PREGEN_JSON=./../grpc_50kb.json scenarios/grpc.js
```
3. Switch shard 0 to disabled mode on the fly (change config and use SIGHUP)
4. k6 load had a lot of errors
```
INFO[0047] status: code = 2049 message = object not found  source=console
```

**Logs:** [bug_shard_errors (2).zip](https://github.com/nspcc-dev/neofs-node/files/9765148/bug_shard_errors.2.zip)
**Versions:**
```
NeoFS Storage node
Version: v0.32.0-91-g807c0a13
GoVersion: go1.18.4
```
## Your Environment
Server setup and configuration:
cloud, 4 VMs, 4 SN, 4 http qw, 4 s3 gw

Operating System and version (uname -a):
linux vedi 5.10.0-16-amd64 https://github.com/nspcc-dev/neofs-node/issues/1 SMP Debian 5.10.127-1 (2022-06-30) x86_64 GNU/Linux

**Comment by fyrchik** [comment]
This is also worth investigating
`INFO[0082] status: code = 1024 message = incomplete object PUT by placement: could not close object stream: (*putsvc.remoteTarget) could not put object to [/dns4/node2.neofs/tcp/8080]: write object via client: context deadline exceeded  source=console`

**Comment by anikeev-yadro** [comment]
> This is also worth investigating `INFO[0082] status: code = 1024 message = incomplete object PUT by placement: could not close object stream: (*putsvc.remoteTarget) could not put object to [/dns4/node2.neofs/tcp/8080]: write object via client: context deadline exceeded source=console`

I think this is related to 50 clients in k6 load. Because it was not reproduced if I used 25 clients.

**Comment by cthulhu-rider** [comment]
Do we have address of the object which have been lost?

There are errors in log:
```
Oct 12 12:14:26 az neofs-node[12530]: 2022-10-12T12:14:26.566Z        debug        get/remote.go:34        remote call failed        {"component": "Object.Get service", "request": "GET", "address": "331CCw5pEseWszpCuDLiFoeCaqiATzEpMzHqC5hVa4vA/7CwN7nU7uWjdkT2wdycRptAEPbovMom4iyHU9XyU9wXX", "raw": false, "local": false, "with session": true, "with bearer": false, "error": "reading the response failed: context deadline exceeded"}
```
Has the other node been loaded too? I guess it might not return an object due to overload.

**Comment by anikeev-yadro** [comment]
@cthulhu-rider Unfortunately no. Because I thought that is clear from logs. 
Other node doesn't have any load. 
Regarding of degraded - I switched shard to disabled, not degraded.

**Comment by cthulhu-rider** [comment]
@anikeev-yadro can you gather and provide log of the other storage node? I could not reproduce the problem without the load.

> I think this is related to 50 clients in k6 load. Because this error was not reproduced if I used 25 clients.

In `az` log I see only `context deadline` errors which finally lead to overall `object not found`. The main hypothesis is in the overload (or stagnation) of the second storage node, because `az` redirects part of the load to it.

**Comment by anikeev-yadro** [comment]
@cthulhu-rider I had to reproduce today on ```NeoFS node Version: v0.32.0-121-gade19077```, because I reinstalled environment.
New logs from
Node az: [bug_shard_errors_az.zip](https://github.com/nspcc-dev/neofs-node/files/9798157/bug_shard_errors_az.zip)
Node buki: [bug_shard_errors_buki.zip](https://github.com/nspcc-dev/neofs-node/files/9798158/bug_shard_errors_buki.zip)
k6 prefill:
```
admin@perf-load-01:~/xk6$ ./scenarios/preset/preset_grpcREP2-Russia.py --size 10000 --containers 5 --out grpc_50kb.json --endpoint 172.26.162.147:8080 --preload_obj 5
Namespace(size='10000', containers='5', out='grpc_50kb.json', preload_obj='5', endpoint='172.26.162.147:8080', update=None)
Create containers: 5
Create containers: Completed
 > Containers: ['D96Royh5vHCMZaYXAKSpo5xFQiHTPf6v2181tyvTGmVT', '8jBjxGsfPwF6ZaEz58jzRDVCmZMi9f94nGRn3q3HGiZK', '32Yvb6kADTGKSP1k74nCnh8SKb6ooKnUGsHKupfBQCQs', 'CmPceyNoNXacrXwog8uYnbZ9j7EwTXu2MvE6smUrpPHa', '3Zz9hNp8SV6Bafo3njfxHVcHnWT9vqw7JhSeLyRKDycy']
Upload objects to each container: 5
 > Create random payload: Completed
 > Upload objects for container D96Royh5vHCMZaYXAKSpo5xFQiHTPf6v2181tyvTGmVT
 > Upload objects for container D96Royh5vHCMZaYXAKSpo5xFQiHTPf6v2181tyvTGmVT: Completed
 > Upload objects for container 8jBjxGsfPwF6ZaEz58jzRDVCmZMi9f94nGRn3q3HGiZK
 > Upload objects for container 8jBjxGsfPwF6ZaEz58jzRDVCmZMi9f94nGRn3q3HGiZK: Completed
 > Upload objects for container 32Yvb6kADTGKSP1k74nCnh8SKb6ooKnUGsHKupfBQCQs
 > Upload objects for container 32Yvb6kADTGKSP1k74nCnh8SKb6ooKnUGsHKupfBQCQs: Completed
 > Upload objects for container CmPceyNoNXacrXwog8uYnbZ9j7EwTXu2MvE6smUrpPHa
 > Upload objects for container CmPceyNoNXacrXwog8uYnbZ9j7EwTXu2MvE6smUrpPHa: Completed
 > Upload objects for container 3Zz9hNp8SV6Bafo3njfxHVcHnWT9vqw7JhSeLyRKDycy
 > Upload objects for container 3Zz9hNp8SV6Bafo3njfxHVcHnWT9vqw7JhSeLyRKDycy: Completed
Upload objects to each container: Completed
Result:
 > Total Containers has been created: 5.
 > Total Objects has been created: 25.
```
k6 log:
```
admin@perf-load-01:~/xk6$ grep REP scenarios/preset/preset_grpcREP2-Russia.py
    policy = "REP 2 IN X SELECT 2 FROM C AS X FILTER Country EQ 'Russia' AS C"
admin@perf-load-01:~/xk6$ date;./k6 run -e PROFILE=50:240 -e WRITE_OBJ_SIZE=10000 -e CLIENTS=25 -e NODES=172.26.162.147:8080 -e PREGEN_JSON=./../grpc_50kb.json scenarios/grpc.js
Пн 17 окт 2022 07:47:52 UTC

          /\      |‾‾| /‾‾/   /‾‾/
     /\  /  \     |  |/  /   /  /
    /  \/    \    |     (   /   ‾‾\
   /          \   |  |\  \ |  (‾)  |
  / __________ \  |__| \__\ \_____/ .io

  execution: local
     script: scenarios/grpc.js
     output: -

  scenarios: (100.00%) 2 scenarios, 25 max VUs, 4m5s max duration (incl. graceful stop):
           * read: 13 looping VUs for 4m0s (exec: obj_read, gracefulStop: 5s)
           * write: 12 looping VUs for 4m0s (exec: obj_write, gracefulStop: 5s)

INFO[0000] Pregenerated containers: 5                    source=console
INFO[0000] Pregenerated read object size: 10000 Kb       source=console
INFO[0000] Pregenerated total objects: 25                source=console
INFO[0027] status: code = 2049 message = object not found  source=console
INFO[0028] status: code = 2049 message = object not found  source=console
INFO[0028] status: code = 2049 message = object not found  source=console
INFO[0028] status: code = 2049 message = object not found  source=console
INFO[0029] status: code = 2049 message = object not found  source=console
INFO[0029] status: code = 2049 message = object not found  source=console
INFO[0030] status: code = 2049 message = object not found  source=console
INFO[0031] status: code = 2049 message = object not found  source=console
INFO[0031] status: code = 2049 message = object not found  source=console
INFO[0032] status: code = 2049 message = object not found  source=console
INFO[0032] status: code = 2049 message = object not found  source=console
INFO[0032] status: code = 2049 message = object not found  source=console
INFO[0035] status: code = 2049 message = object not found  source=console

running (4m02.4s), 00/25 VUs, 2143 complete and 0 interrupted iterations
read  ✓ [======================================] 13 VUs  4m0s
write ✓ [======================================] 12 VUs  4m0s

     █ setup

     data_received............: 19 GB  78 MB/s
     data_sent................: 3.0 GB 12 MB/s
     iteration_duration.......: avg=2.81s min=125.41µs med=1.29s max=35.22s p(90)=5.69s  p(95)=15.67s
     iterations...............: 2143   8.840269/s
     neofs_obj_get_duration...: avg=1.59s min=62.7ms   med=1.16s max=35.22s p(90)=2.15s  p(95)=2.69s
     neofs_obj_get_fails......: 13     0.053627/s
     neofs_obj_get_total......: 1848   7.62334/s
     neofs_obj_put_duration...: avg=9.83s min=1.75s    med=6.14s max=26.03s p(90)=21.54s p(95)=23.08s
     neofs_obj_put_total......: 295    1.216929/s
     vus......................: 7      min=7      max=25
     vus_max..................: 25     min=25     max=25
```
k6 prefill objects:
```
{"containers": ["D96Royh5vHCMZaYXAKSpo5xFQiHTPf6v2181tyvTGmVT", "8jBjxGsfPwF6ZaEz58jzRDVCmZMi9f94nGRn3q3HGiZK", "32Yvb6kADTGKSP1k74nCnh8SKb6ooKnUGsHKupfBQCQs", "CmPceyNoNXacrXwog8uYnbZ9j7EwTXu2MvE6smUrpPHa", "3Zz9hNp8SV6Bafo3njfxHVcHnWT9vqw7JhSeLyRKDycy"], "objects": [{"container": "D96Royh5vHCMZaYXAKSpo5xFQiHTPf6v2181tyvTGmVT", "object": "6Xrph6WNiURLXfJ2uZ4sXcyUv67nZ3uozeyDZDrMoe27"}, {"container": "D96Royh5vHCMZaYXAKSpo5xFQiHTPf6v2181tyvTGmVT", "object": "48bx8h5pqT96ystBc126Cto8h9U8FtW6vnjpxpfWtecU"}, {"container": "D96Royh5vHCMZaYXAKSpo5xFQiHTPf6v2181tyvTGmVT", "object": "9RgsbwMmBCyU87YKbCcz5zUPYcrvBSMesqJ1eVFbek6s"}, {"container": "D96Royh5vHCMZaYXAKSpo5xFQiHTPf6v2181tyvTGmVT", "object": "HHRuxX1JPnaLrRBaKBTFbWw9k7rcUZwAAZLrB7m1fJ57"}, {"container": "D96Royh5vHCMZaYXAKSpo5xFQiHTPf6v2181tyvTGmVT", "object": "7qReiDPGyyzABURXq9bQPBhdnw7TjUd25AMxB9jn5d3H"}, {"container": "8jBjxGsfPwF6ZaEz58jzRDVCmZMi9f94nGRn3q3HGiZK", "object": "Co6wNdeMUctp7LSJvhUKEy3kf3hqAh1d3sFAEFXBGi3p"}, {"container": "8jBjxGsfPwF6ZaEz58jzRDVCmZMi9f94nGRn3q3HGiZK", "object": "8fL9RFA9vmZU8ppdyrUd8vx76DcMVvwLvNMXRn4Y35LW"}, {"container": "8jBjxGsfPwF6ZaEz58jzRDVCmZMi9f94nGRn3q3HGiZK", "object": "4xQRyc8a8GgC299vgtRwwWMLi1NqXFQqs9ccNk5NvyPp"}, {"container": "8jBjxGsfPwF6ZaEz58jzRDVCmZMi9f94nGRn3q3HGiZK", "object": "Eg9q1srUR9oTtUEKUvvHs218EGgMqM1L237QDbBcn2eV"}, {"container": "8jBjxGsfPwF6ZaEz58jzRDVCmZMi9f94nGRn3q3HGiZK", "object": "5dDPypQHJ5RZEuqdjDd3K3ZWFrcbhcQ8Edgm9uDHBDoR"}, {"container": "32Yvb6kADTGKSP1k74nCnh8SKb6ooKnUGsHKupfBQCQs", "object": "4QxEtrXsWDAMsu7txnBerDE6XKZ1qNXmccuHAaP8VvQ9"}, {"container": "32Yvb6kADTGKSP1k74nCnh8SKb6ooKnUGsHKupfBQCQs", "object": "3v2XbkVTbWp5LBdaQoW8Vwng46BrAskiAruvUW8MNTz4"}, {"container": "32Yvb6kADTGKSP1k74nCnh8SKb6ooKnUGsHKupfBQCQs", "object": "5DpBpxRr6dzAGS9PDa3yDEqdbjJbcsM6dcr9wuR3PzVi"}, {"container": "32Yvb6kADTGKSP1k74nCnh8SKb6ooKnUGsHKupfBQCQs", "object": "EiuaBXWbtiGGkjNuib6tPtvnxRCnSytcHQK6bJjYT1Zk"}, {"container": "32Yvb6kADTGKSP1k74nCnh8SKb6ooKnUGsHKupfBQCQs", "object": "2roUDFTRum2mfKN4enJrfrMC17z5Wn6A2usrfcpSzT6e"}, {"container": "CmPceyNoNXacrXwog8uYnbZ9j7EwTXu2MvE6smUrpPHa", "object": "3DuSXLJ7xRTevSeCzMWZtSS5YmLa1eNZr6KVU1xM3Gmm"}, {"container": "CmPceyNoNXacrXwog8uYnbZ9j7EwTXu2MvE6smUrpPHa", "object": "4918326ZYeHDjMF1w9eScdEcfnoFPFj2bcKDtDsNScre"}, {"container": "CmPceyNoNXacrXwog8uYnbZ9j7EwTXu2MvE6smUrpPHa", "object": "FJRnHJDZEJQrfN5CpnwsFQDCoXrtXCRCy4SkuLhPkfyp"}, {"container": "CmPceyNoNXacrXwog8uYnbZ9j7EwTXu2MvE6smUrpPHa", "object": "DZbZ5FDbMuGdiFNaqoTdGrquzV9jdyGp87DzigQNM1YZ"}, {"container": "CmPceyNoNXacrXwog8uYnbZ9j7EwTXu2MvE6smUrpPHa", "object": "ETykN2WtqJ54Gtvr9aozgUUEYcxz84g5CF4iqRD7BMtT"}, {"container": "3Zz9hNp8SV6Bafo3njfxHVcHnWT9vqw7JhSeLyRKDycy", "object": "8h6Mo8FBJToviqDpQEyJsrKgAvevgEBZnWFKWjo38vzb"}, {"container": "3Zz9hNp8SV6Bafo3njfxHVcHnWT9vqw7JhSeLyRKDycy", "object": "AypoiM74avKBSFTK8bKef4Td6kFoQEhN3nuNhndfNjhg"}, {"container": "3Zz9hNp8SV6Bafo3njfxHVcHnWT9vqw7JhSeLyRKDycy", "object": "EEmtxhVnrkMcmDopArFSiKA8P4YwRbeMZbhdLbpKJ7st"}, {"container": "3Zz9hNp8SV6Bafo3njfxHVcHnWT9vqw7JhSeLyRKDycy", "object": "HymifXRpd2nNZSucGg1aeXeZywvFgNkG43huPXQKotWV"}, {"container": "3Zz9hNp8SV6Bafo3njfxHVcHnWT9vqw7JhSeLyRKDycy", "object": "5DHBMAbm2Dw26v2aLz1eQ98TnXqfj8CKENdFyi6M9xgB"}], "obj_size": "10000 Kb"}
```

**Comment by fyrchik** [comment]
There are many `context exceeded` errors in policer while trying to replicate missing copies.
This should be fixed with #2049 , can we check this now? @anikeev-yadro
